precompiles/dapps-staking/src/lib.rs

```diff
@@ -48,159 +50,122 @@ where
         let staked = reward_and_stake.map_or(Zero::zero(), |r| r.staked);
         let staked = TryInto::<u128>::try_into(staked).unwrap_or(0);
-        let output = EvmDataWriter::new().write(staked).build();
-
-        Ok(PrecompileOutput {
-            exit_status: ExitSucceed::Returned,
-            cost: gasometer.used_gas(),
-            output,
-            logs: Default::default(),
-        })
+
+        Ok(succeed(EvmDataWriter::new().write(staked).build()))
     }
 
     /// Fetch Ledger storage map for an account
-    fn read_staked_amount(
-        input: &mut EvmDataReader,
-        gasometer: &mut Gasometer,
-    ) -> EvmResult<PrecompileOutput> {
-        input.expect_arguments(gasometer, 1)?;
+    fn read_staked_amount(handle: &mut impl PrecompileHandle) -> EvmResult<PrecompileOutput> {
+        handle.record_cost(RuntimeHelper::<R>::db_read_gas_cost())?;
+
+        let mut input = handle.read_input()?;
+        input.expect_arguments(1)?;
 
         // parse input parameters for pallet-dapps-staking call
-        let staker_vec: Vec<u8> = input.read::<Bytes>(gasometer)?.into();
-        let staker = Self::parse_input_address(gasometer, staker_vec)?;
+        let x = input.read::<AccountId32>()?;
+        let staker_vec: &[u8; 32] = x.as_ref();
+        let staker = R::AccountId::from(*staker_vec);
 
         // call pallet-dapps-staking
         let ledger = pallet_dapps_staking::Ledger::<R>::get(&staker);
         log::trace!(target: "ds-precompile", "read_staked_amount for account:{:?}, ledger.locked:{:?}", staker, ledger.locked);
 
-        // compose output
-        let output = EvmDataWriter::new().write(ledger.locked).build();
-
-        Ok(PrecompileOutput {
-            exit_status: ExitSucceed::Returned,
-            cost: gasometer.used_gas(),
-            output,
-            logs: Default::default(),
-        })
+        Ok(succeed(EvmDataWriter::new().write(ledger.locked).build()))
     }
 
     /// Read GeneralStakerInfo for account/contract
     fn read_staked_amount_on_contract(
-        input: &mut EvmDataReader,
-        gasometer: &mut Gasometer,
+        handle: &mut impl PrecompileHandle,
     ) -> EvmResult<PrecompileOutput> {
-        input.expect_arguments(gasometer, 2)?;
+        handle.record_cost(RuntimeHelper::<R>::db_read_gas_cost())?;
+
+        let mut input = handle.read_input()?;
+        input.expect_arguments(2)?;
 
         // parse contract address
-        let contract_h160 = input.read::<Address>(gasometer)?.0;
-        let contract_id = Self::decode_smart_contract(gasometer, contract_h160)?;
+        let contract_h160 = input.read::<Address>()?.0;
+        let contract_id = Self::decode_smart_contract(contract_h160)?;
 
         // parse input parameters for pallet-dapps-staking call
-        let staker_vec: Vec<u8> = input.read::<Bytes>(gasometer)?.into();
-        let staker = Self::parse_input_address(gasometer, staker_vec)?;
+        let x = input.read::<AccountId32>()?;
```

Comment:
Read static length value instead of dynamic length could affect result, in solidity it in codes differently.

Reply:
Sorry, just noticed this comment.
Let me check.

I can fix the implementation of reading `AccountId32` type if needed.